pkg/api/resource_types.go

```diff
@@ -12,10 +12,11 @@ import (
 
 type Resource struct {
 	Meta
-	Version    int32
-	ConsumerID string
-	Manifest   datatypes.JSONMap
-	Status     datatypes.JSONMap
+	Version         int32
+	ObservedVersion int32
```

Comment:
Do we need to have `ObservedVersion` in Resource struct? In my understand, it may be in `ReconcileStatus` only.

Reply:
this will  help consumer to known their resource actual version in the agent easily, the consumer will not need to parse the status to get the   resource actual version, e.g. the consumer get a resource from the maestro,  it will get the resource like this

```
id: ...
version: 5
observedVersion: 4
spec: ....
status: ....
```

Reply:
do we also need to update the `ObservedGeneration` to `ObservedVersion` to avoid confusion? per my understanding, the `ObservedGeneration` is also the version from spec, not the actual kube version.